lib/factory.rb

```diff
@@ -14,3 +18,82 @@
 # - to_a
 # - values_at
 # - ==, eql?
+
+class Factory
+  class << self
+    def new(*args, &block)
+      const_set(args.shift.capitalize, create_new_class(*args, &block)) if args.first.is_a?(String)
+      create_new_class(*args, &block)
+    end
+
+    def create_new_class(*args, &block)
+      Class.new do
+        attr_accessor(*args)
+
+        define_method :initialize do |*variables|
+          raise ArgumentError, "Expected #{args.count}" if args.count != variables.count
+
+          args.each_index do |index|
+            instance_variable_set("@#{args[index]}", variables[index])
+          end
+        end
+
+        define_method :each do |&block|
+          values.each(&block)
+        end
+
+        define_method :each_pair do |&pair|
+          to_h.each_pair(&pair)
+        end
+
+        define_method :dig do |*args|
+          args.inject(self) { |values, args| values[args] if values }
+        end
+
+        define_method :to_h do
+          args.zip(values).to_h
+        end
+
+        define_method :length do
+          args.size
+        end
+
+        define_method :members do
+          args
+        end
+
+        define_method :select do |&block|
+          values.select(&block)
+        end
+
+        define_method :eql? do |other|
+          self.class == other.class && values == other.values
+        end
+
+        define_method(:to_a) do
+          instance_variables.map { |argument| instance_variable_get(argument) }
+        end
+
+        define_method(:values_at) do |*index|
```

Comment:
this can be implemented with `select` in one line

Reply:
done